### PR TITLE
feat(connector): [Opayo] Add script generated template code

### PR DESCRIPTION
**Pull request by prasunna09**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Add the template code generated by script for [Opayo](https://pi-test.sagepay.com/api/v1/) with test base url.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/tests/connectors/sample_auth.toml [hunk 1]
```diff
@@ -115,3 +115,7 @@ api_key = "API Key"
 api_key = "Application API KEY"
 api_secret = "Application Identifier"
 key1 = "Business Identifier"
+
```
Comment (srujanchikke) [inline crates/router/tests/connectors/sample_auth.toml:118]: Please remove this newline.

**Review by srujanchikke — DISMISSED** [review]

File: crates/router/src/connector/opayo.rs [hunk 2]
```diff
@@ -0,0 +1,499 @@
...
+        let response: opayo::OpayoPaymentsResponse = res
+            .response
+            .parse_struct("Opayo PaymentsAuthorizeResponse")
+            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+        types::RouterData::try_from(types::ResponseRouterData {
+            response,
+            data: data.clone(),
+            http_code: res.status_code,
+        })
+        .change_context(errors::ConnectorError::ResponseHandlingFailed)
```
Comment (ArjunKarthik) [inline crates/router/src/connector/opayo.rs:189]: change context is not required. If it is added in the template code can we remove there as well?

File: crates/router/src/connector/opayo.rs [hunk 3]
```diff
@@ -0,0 +1,499 @@
...
+        let response: opayo::OpayoPaymentsResponse = res
+            .response
+            .parse_struct("opayo PaymentsSyncResponse")
+            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+        types::RouterData::try_from(types::ResponseRouterData {
+            response,
+            data: data.clone(),
+            http_code: res.status_code,
+        })
+        .change_context(errors::ConnectorError::ResponseHandlingFailed)
```
Comment (ArjunKarthik) [inline crates/router/src/connector/opayo.rs:252]: same here

File: crates/router/src/connector/opayo.rs [hunk 4]
```diff
@@ -0,0 +1,499 @@
...
+        let response: opayo::OpayoPaymentsResponse = res
+            .response
+            .parse_struct("Opayo PaymentsCaptureResponse")
+            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+        types::RouterData::try_from(types::ResponseRouterData {
+            response,
+            data: data.clone(),
+            http_code: res.status_code,
+        })
+        .change_context(errors::ConnectorError::ResponseHandlingFailed)
```
Comment (ArjunKarthik) [inline crates/router/src/connector/opayo.rs:325]: same here

File: crates/router/src/connector/opayo.rs [hunk 5]
```diff
@@ -0,0 +1,499 @@
...
+        let response: opayo::RefundResponse = res
+            .response
+            .parse_struct("opayo RefundResponse")
+            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+        types::RouterData::try_from(types::ResponseRouterData {
+            response,
+            data: data.clone(),
+            http_code: res.status_code,
+        })
+        .change_context(errors::ConnectorError::ResponseHandlingFailed)
```
Comment (ArjunKarthik) [inline crates/router/src/connector/opayo.rs:404]: here also

File: crates/router/src/connector/opayo.rs [hunk 6]
```diff
@@ -0,0 +1,499 @@
...
+        let response: opayo::RefundResponse = res
+            .response
+            .parse_struct("opayo RefundSyncResponse")
+            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+        types::RouterData::try_from(types::ResponseRouterData {
+            response,
+            data: data.clone(),
+            http_code: res.status_code,
+        })
+        .change_context(errors::ConnectorError::ResponseHandlingFailed)
```
Comment (ArjunKarthik) [inline crates/router/src/connector/opayo.rs:466]: .

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by jagan-jaya — APPROVED** [review]

**Review by jagan-jaya — APPROVED** [review]

**Review by ArjunKarthik — DISMISSED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by srujanchikke — APPROVED** [review]

**Review by ArjunKarthik — DISMISSED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by srujanchikke — DISMISSED** [review]

**Review by SamraatBansal — DISMISSED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

File: crates/router/src/connector/opayo/transformers.rs [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+                })
+            }
+            _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
+        }
+    }
+}
+
+// Auth Struct
+pub struct OpayoAuthType {
+    pub(super) api_key: String,
```
Comment (SanchithHegde) [inline crates/router/src/connector/opayo/transformers.rs:51]: Would prefer if this was `Secret`.

File: crates/router/src/connector/opayo.rs [hunk 8]
```diff
@@ -87,7 +88,7 @@ impl ConnectorCommon for Opayo {
             .change_context(errors::ConnectorError::FailedToObtainAuthType)?;
         Ok(vec![(
             headers::AUTHORIZATION.to_string(),
-            auth.api_key.into_masked(),
+            auth.api_key.peek().to_string().into_masked(),
```
Comment (SanchithHegde) [inline crates/router/src/connector/opayo.rs:91]: Can you use `expose()` instead?

```suggestion
            auth.api_key.expose().into_masked(),
```

You'll have to import `ExposeInterface` by the way.

**Review by srujanchikke — APPROVED** [review]

**Review by jagan-jaya — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**